omp/components/prefix_sum_kernels.cpp

```diff
@@ -68,6 +70,8 @@ void prefix_sum(std::shared_ptr<const OmpExecutor> exec,
     const int nthreads = omp_get_max_threads();
     vector<IndexType> proc_sums(nthreads, 0, {exec});
     const size_type def_num_witems = (num_entries - 1) / nthreads + 1;
+    bool overflow = false;
+    constexpr auto max = std::numeric_limits<IndexType>::max();
```

Comment:
the implementation will give slightly different reult in the corner case

Reply:
can you elaborate?

Reply:
I mean the exception actually.
size_type in cuda/hip: if the value is max, it will throw the error. it won't throw in the omp/reference.
signed value: if the first value is negative and the second is positive (the first summation is positive), it will not throw the error in omp/reference